lib/checks/navigation/heading-order-evaluate.js

```diff
@@ -4,17 +4,18 @@ import { isVisible } from '../../commons/dom';
 
 function getLevel(vNode) {
   const role = vNode.attr('role');
-  if (role && role.includes('heading')) {
-    const ariaHeadingLevel = vNode.attr('aria-level');
-    const level = parseInt(ariaHeadingLevel, 10);
-
-    // default aria-level for a heading is 2 if it is
-    // not set or set to an incorrect value
-    // @see https://www.w3.org/TR/wai-aria-1.1/#heading
-    if (isNaN(level) || level < 1 || level > 6) {
-      return 2;
-    }
+  const headingRole = role && role.includes('heading');
+  const ariaHeadingLevel = vNode.attr('aria-level');
+  const level = parseInt(ariaHeadingLevel, 10);
+
+  // default aria-level for a heading is 2 if it is
+  // not set or set to an incorrect value
+  // @see https://www.w3.org/TR/wai-aria-1.1/#heading
+  if ((headingRole && isNaN(level)) || level < 1 || level > 6) {
```

Comment:
This is going to fail on `<h1 role="heading">`. Not sure if that's a new problem, but it's a problem.